declarations_site/spotter/utils.py

```diff
@@ -255,3 +267,49 @@ def create_notify(task, report):
     send_found_notify(notify)
     notify.save()
     return notify
+
+
+def save_search_task(user, query, deepsearch=True, query_params='', chat_data=''):
+    if len(query) < 2:
+        raise ValueError('Не вдалось створити завдання з пустим запитом.')
+
+    if len(query) > 100 or len(query.split(' ')) > 10:
+        raise ValueError('Не вдалось створити завдання з таким довгим запитом.')
+
+    if not user.email:
+        raise ValueError('Не вдалось створити завдання без адреси електронної пошти.')
+
+    # don't add twice
+    if SearchTask.objects.filter(user=user, query=query,
+            deepsearch=deepsearch, query_params=query_params, is_deleted=False).exists():
+        raise ValueError('Таке завдання вже існує.')
+
+    task = SearchTask(user=user, query=query, deepsearch=deepsearch)
+    task.query_params = query_params
+    task.chat_data = chat_data
+    task.save()
+
+    if not first_run(task):
+        task.is_deleted = True
+        task.save()
+        raise ValueError('Не вдалось створити завдання, спробуйте спростити запит "%s"' % task.query)
+
+    return task
+
+
+def find_search_task(user, query, deepsearch=True, query_params=''):
+    tasks = SearchTask.objects.filter(user=user, query=query, deepsearch=deepsearch,
+                                query_params=query_params, is_deleted=False)[:1]
+    for t in tasks:
+        return t
+
+
+def list_search_tasks(user):
+    return SearchTask.objects.filter(user=user, is_deleted=False)
+
+
+def get_user_notify(notify_id, **kwargs):
+    try:
+        return NotifySend.objects.filter(id=notify_id, **kwargs)[0]
```

Comment:
Erm, you are filtering it by primary key here, why should it return more than 1 result?